### PR TITLE
Fix scrolling when game speed was increased.

### PR DESCRIPTION
**Pull request by ZehM4tt**
One of my recent changes moved the input handling into game_update_logic which is ran per tick, input should be managed per frame instead.

Also introduced a new global variable that will prevent the scenario_rand to assert because gInUpdateCode is not set at that time.

I've also added to set gInUpdateCode at few missing places that would cause it to assert while it shouldn't.

### CODE REVIEW

**Comment by janisozaur** [comment]
@ZehM4tt please keep PRs single-topic

**Comment by ZehM4tt** [comment]
What do you mean?

**Comment by janisozaur** [comment]
Don't include unrelated changes to UI and logic in single commit.

**Comment by ZehM4tt** [comment]
They are not quite unrelated, I had to move gInUpdateCode otherwise it would assert, once I did that it started to assert in the network.cpp, once I had that it started to assert somewhere else etc. So it all came down to whats in the commit.

File: src/openrct2/game.c [hunk 1]
```diff
@@ -1420,7 +1410,7 @@ void game_load_or_quit_no_save_prompt()
  */
 void game_init_all(sint32 mapSize)
 {
-    gInUpdateCode = true;
+    gInMapInitCode = true;
```
Comment (janisozaur) [inline src/openrct2/game.c]: Why can't it stay as is?

Comment (janisozaur) [inline src/openrct2/game.c]: You still end up checking both of these variables, so it would look like you can just leave it the way it was.

**Comment by janisozaur** [comment]
@ZehM4tt I admit I don't fully understand the purpose of this change. Why is setting of `gInUpdateCode` moved out of `game_update_logic`? It also now includes input handling, which I don't think is right, unless you convince me otherwise.

Can you please provide some explanation?

**Comment by ZehM4tt** [comment]
I'm not entirely sure why so many asserts are there that check this. One problem with the old way is that game_init_all did set it to false afterwards, so later within the update it would have been false, this caused map loading to always assert.

I also see no harm in wrapping the input handling, outgoing commands are enqueued so there is nothing wrong to issue commands within input handling now. It is possible that the assert is a old left over as when I started there was only one ifdef for DEBUG_DESYNC which was the assert in scenario_rand.

I did test the changes today with another mate, so far we did not encounter a single desync with the recent PRs.

**Pull request merged**